sdk/core/core/CMakeLists.txt

```diff
@@ -51,26 +51,28 @@ target_include_directories (az_core PUBLIC inc)
 # make sure that users can consume the project as a library.
 add_library (az::core ALIAS az_core)
 
-add_executable (az_core_test test/main.c)
+# build test and executables only if requested
+if (BUILD_AZ_CORE)
+  add_executable (az_core_test test/main.c)
+  target_link_libraries(az_core_test PRIVATE az_core)
+  target_link_libraries(az_core PRIVATE CURL::libcurl)
 
-target_link_libraries(az_core_test PRIVATE az_core)
+  # curl easy perform POC
+  add_executable (curl_easy_perform_poc test/curl_easy_perform_poc.c)
+  target_link_libraries(curl_easy_perform_poc PRIVATE 
+    az_core 
+    CURL::libcurl)
 
-# curl easy perform POC
-add_executable (curl_easy_perform_poc test/curl_easy_perform_poc.c)
-target_link_libraries(curl_easy_perform_poc PRIVATE 
-  az_core 
-  CURL::libcurl)
+  if (UNIX)
```

Comment:
UNIX is the wrong thing to check here, but the right thing to check would be really hard so it's fine (the correct thing to check is which C standard library implementation you're using).